src/Simulations/simulation.jl

```diff
@@ -69,7 +69,10 @@ function Simulation(model; Δt,
              "recalculate the time step every iteration which can be slow."
    end
 
-   diagnostics[:nan_checker] = NaNChecker(fields=(u=model.velocities.u,),
+   # Check for NaNs in the model's first field.
+   model_fields = fields(model)
+   field_to_check_nans = NamedTuple{(keys(model_fields)[1],)}((model_fields[1],))
```

Comment:
```suggestion
   field_to_check_nans = NamedTuple{tuple(first(keys(model_fields)))}(first(model_fields))
```

not sure it's better but might be.

Reply:
You missed a `tuple` but I think it's definitely better! Made even better/more readable with pipes:

```julia
NamedTuple{keys(model_fields) |> first |> tuple}(first(model_fields) |> tuple)
```

Reply:
It's not relevant here but `first` is more general since it can work for iterables, in addition to concrete collections (i think)